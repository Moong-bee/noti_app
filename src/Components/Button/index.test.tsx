import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import 'jest-styled-components'

import { Button } from './index'

describe('<Button />', () => {
  it('올바른 랜더링', () => {
    const { container } = render(<Button label="버튼" />)

    const label = screen.getByText('버튼')
    expect(label).toBeInTheDocument()

    const parent = label.parentElement
    expect(parent).toHaveStyleRule('background-color', '#0077b6')
    expect(parent).toHaveStyleRule('background-color', '#007cbd', {
      modifier: ':hover',
    })

    expect(container).toMatchSnapshot()
  })

  it('background color와 hover color 변경', () => {
    const backgroundColor = '#d00000'
    const hoverColor = '#e00000'
    render(<Button label="삭제" backgroundColor={backgroundColor} hoverColor={hoverColor} />)

    const parent = screen.getByText('삭제').parentElement
    expect(parent).toHaveStyleRule('background-color', backgroundColor)
    expect(parent).toHaveStyleRule('background-color', hoverColor, {
      modifier: ':hover',
    })
  })

  it('버튼 클릭', () => {
    const handleClick = jest.fn()
    render(<Button label="버튼" onClick={handleClick} />)

    const button = screen.getByText('버튼')
    expect(handleClick).toHaveBeenCalledTimes(0)
    fireEvent.click(button)
    expect(handleClick).toHaveBeenCalledTimes(1)
  })
})
