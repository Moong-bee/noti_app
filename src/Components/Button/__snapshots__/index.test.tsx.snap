// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> 올바른 랜더링 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 10px 20px;
  cursor: pointer;
  border-radius: 8px;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.2);
  background-color: #0077b6;
}

.c0:hover {
  background-color: #007cbd;
}

.c0:active {
  box-shadow: inset 5px 5px 10px rgba(0,0,0,0.2);
}

.c1 {
  color: #FFFFFF;
  font-size: 16px;
}

<div>
  <div
    class="c0"
    type="fill"
  >
    <div
      class="c1"
      type="fill"
    >
      버튼
    </div>
  </div>
</div>
`;
